app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/DisplayImageSetFragment.kt

```diff
@@ -114,29 +116,131 @@ class DisplayImageSetFragment: Fragment() {
+                            }
+                            categorizedPicturesSelectedList = emptySet()
+                            numberOfSelectedPictures = 0
+                            mode!!.finish()
+                        }
+                    R.id.set_representative_picture ->
+                        lifecycleScope.launch {
+                            val pic = categorizedPicturesSelectedList.first()
+                            dBManagement.putRepresentativePicture(
+                                pic
+                            )
+                            categorizedPicturesList = categorizedPicturesList.minus(pic)
+                            categorizedPicturesSelectedList = emptySet()
+                            numberOfSelectedPictures = 0
+                            mode!!.finish()
+                        }
+                    else -> {
+
+                    }
+                }
+                return true
+            }
+
+            override fun onDestroyActionMode(mode: ActionMode?) {
+                for (i in categorizedPicturesList.indices) {
+                    binding.displayImageSetImagesGridView[i].alpha = 1F
+                    binding.displayImageSetImagesGridView[i].elevation = tenDp
+                }
+                displayImageSetAdapter.updatePictures(categorizedPicturesList)
+                displayImageSetAdapter.notifyDataSetChanged()
+            }
+
+            override fun onItemCheckedStateChanged(
+                mode: ActionMode?,
+                position: Int,
+                id: Long,
+                checked: Boolean
+            ) {
+                if (checked) {
+                    numberOfSelectedPictures += 1
+                    mode!!.title = "$numberOfSelectedPictures selected pictures"
+                    binding.displayImageSetImagesGridView[position].alpha = 0.35F
+                    binding.displayImageSetImagesGridView[position].elevation = 0F
+                    categorizedPicturesSelectedList =
+                        categorizedPicturesSelectedList.plus(
+                            categorizedPicturesList.elementAt(
+                                position
+                            )
+                        )
+                } else {
+                    numberOfSelectedPictures -= 1
+                    binding.displayImageSetImagesGridView[position].alpha = 1F
+                    binding.displayImageSetImagesGridView[position].elevation = tenDp
+                    mode!!.title = "$numberOfSelectedPictures selected pictures"
+                    categorizedPicturesSelectedList =
+                        categorizedPicturesSelectedList.minus(
+                            categorizedPicturesList.elementAt(
+                                position
+                            )
+                        )
+                }
+
+                mode.menu[1].isVisible = numberOfSelectedPictures == 1
```

Comment:
What does this line of code exactly do ?

Reply:
This is to set the "Set representative picture " button (the star) invisible when you select more than 1 picture. (Because you can only set 1 picture as representative). You can test it yourself to see the effect whe you select 1 picture and what happens when you selec 2 pictures. ;)

Reply:
Yes I see, and the actionModes represent those various actions as in set representative, representation  ..

Reply:
Are you talking about the "GridView.CHOICE_MODE_MULTIPLE_MODAL" ? This is just a mode for gridviews to be able to select multiple views easily at the same time :)

Reply:
Yes Okay Understood thanks :))